### ISSUE TITLE
[DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated

### ISSUE THREAD

**Issue by alexcastillo** [brightcove/hot-shots#111]
Hi,

Great project, thank you.

I'm getting a deprecation warning on node 12.2.0:

```
10:32:42 0|api      | (node:46081) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
10:32:42 0|api      | Warning DeprecationWarning: Socket.prototype._handle is deprecated
10:32:42 0|api      |     at Object.extractSocket (/Users/alexcastillo/.npm/cache/_npx/45921/lib/node_modules/pm2/node_modules/event-loop-inspector/utils.js:24:23)
10:32:42 0|api      |     at /Users/alexcastillo/.npm/cache/_npx/45921/lib/node_modules/pm2/node_modules/event-loop-inspector/index.js:39:17
10:32:42 0|api      |     at Array.forEach (<anonymous>)
10:32:42 0|api      |     at Object.dump (/Users/alexcastillo/.npm/cache/_npx/45921/lib/node_modules/pm2/node_modules/event-loop-inspector/index.js:22:42)
10:32:42 0|api      |     at Timeout._onTimeout (/Users/alexcastillo/.npm/cache/_npx/45921/lib/node_modules/pm2/node_modules/@pm2/io/build/main/transports/IPCTransport.js:34:50)
10:32:42 0|api      |     at listOnTimeout (internal/timers.js:531:17)
10:32:42 0|api      |     at processTimers (internal/timers.js:475:7) {
10:32:42 0|api      |   name: 'DeprecationWarning',
10:32:42 0|api      |   code: 'DEP0112'
10:32:42 0|api      | }
```
It appers to be related to dgram dependency. Any advice?

**Comment by bdeitte** [comment]
Oh interesting, so unix-dgram installs for you on Node 12.  I didn't even see it install properly previously when I tried it: https://github.com/brightcove/hot-shots/pull/107

This issue shouldn't be a problem for you if you aren't using the brand new 'uds' protocol.  The library mentioned above is not in use at all in non-uds cases.

In order to support future versions of Node where this deprecation may turn into a real error, plus to just get rid of this warning, it would be nice to have the unix-dgram library upgraded to use a non-deprecated API (and then of course use that new version here).  Happy to take a PR from this on anyone, and I will keep this issue open until this is done.

**Comment by thisromantelecom** [comment]
2|index    | (node:20220) [DEP0112] DeprecationWarning: Socket.prototype._handle is deprecated
UDP linux

**Comment by bdeitte** [comment]
This should be fixed in 6.5.0- let me know if you see any issues here.  I fixed it by only loading in the uds library when it's really needed, so this is only seen by a few people now.
